<launch>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find planner)/cfg/2D_map.rviz"/>

  <node pkg="planner" name="trajectory_generation" type="trajectory_generation_node" output="screen">
    <param name="trajectory_generator/map_x_size" value="28.0"/>
    <param name="trajectory_generator/map_y_size" value="15.2"/>  <!-- 仿真环境参数-->
    <!-- <param name="trajectory_generator/map_y_size" value="30.0"/> -->
    <param name="trajectory_generator/map_z_size" value="2.0"/>
    <param name="trajectory_generator/search_height_min" value="-0.3"/>  <!-- 仿真环境参数, 雷达在车头上-->
    <!-- <param name="trajectory_generator/search_height_min" value="0.1"/> -->
    <!-- 搜索高度 -->
    <param name="trajectory_generator/search_height_max" value="1.2"/>
    <!-- 搜索半径 -->
    <param name="trajectory_generator/search_radius" value="8.0"/>
    <param name="trajectory_generator/obstacle_swell_flag" value="true"/>
    <param name="trajectory_generator/robot_radius_dash" value="0.35"/>
    <!-- 机器人半径 -->
    <param name="trajectory_generator/robot_radius" value="0.35"/>
    <param name="trajectory_generator/isxtl" value="false"/>
    <param name="trajectory_generator/xtl_flag" value="false"/>
    <param name="trajectory_generator/map_resolution" value="0.1"/>

    <param name="trajectory_generator/reference_v_max" value="3.5"/>
    <param name="trajectory_generator/reference_a_max" value="10.0"/>
    <param name="trajectory_generator/reference_w_max" value="4.0"/>
    <param name="trajectory_generator/reference_desire_speed" value="2.6"/>  <!-- 区赛速度暂定2.5，仿真环境极限速度3.4，可能实际更难达到 -->
    <param name="trajectory_generator/reference_desire_speedxtl" value="3.0"/>
    <param name="trajectory_generator/reference_vxtl_max" value="1.6"/>
    <param name="trajectory_generator/reference_axtl_max" value="2.0"/>
    <param name="trajectory_generator/reference_wxtl_max" value="2.0"/>

    <param name="trajectory_generator/map_lower_point_x" value="0.0"/>
    <param name="trajectory_generator/map_lower_point_y" value="0.0"/>
    <param name="trajectory_generator/map_lower_point_z" value="0.0"/>

    <param name="trajectory_generator/height_bias" value="-0.5"/>  <!-- 地图高度偏置参数 -->
    <param name="trajectory_generator/height_interval" value="2.0"/> <!-- 地图高度间隔参数，地图所有像素的高度间隔 -->
    <param name="trajectory_generator/height_threshold" value="0.1"/> <!-- 地图高度阈值参数，障碍物点云高度滞环检测阈值 -->
    <param name="trajectory_generator/height_sencond_high_threshold" value="0.3"/> <!-- 桥梁地图高度阈值参数，这个是顶部高度阈值 -->

    <!-- <param name="trajectory_generator/occ_file_path" type="string" value="/home/hitcrt/sentry_ws/src/sentry_planning/trajectory_generation/map/occ506.png"/> -->
    <!-- <param name="trajectory_generator/bev_file_path" type="string" value="/home/hitcrt/sentry_ws/src/sentry_planning/trajectory_generation/map/bev506.png"/> -->
    <!-- <param name="trajectory_generator/distance_map_file_path" type="string" value="/home/hitcrt/sentry_ws/src/sentry_planning/trajectory_generation/map/occtopo506.png"/> -->

    <param name="trajectory_generator/occ_file_path" type="string" value="$(find planner)/map/occ2024.png"/>
    <param name="trajectory_generator/bev_file_path" type="string" value="$(find planner)/map/bev2024low.png"/>
    <param name="trajectory_generator/distance_map_file_path" type="string" value="$(find planner)/map/occtopo2024low.png"/>

  </node>

  <node pkg="planner" name="waypoint_generator" type="waypoint_generator_node" output="screen">
    <remap from="~goal" to="/goal"/>
    <param name="waypoint_type" value="manual-lonely-waypoint"/>
  </node>

</launch>
